src/com/github/Kraken3/AFKPGC/BotDetector.java

```diff
@@ -0,0 +1,306 @@
+			// find new top suspects
+			for (Map.Entry<UUID, LastActivity> entry : lastActivities.entrySet()) {
+				UUID playerUUID = entry.getKey();
+				Player p = Bukkit.getPlayer(playerUUID);
+				/* according to the author of AFKGPC, there might be
+				 * inconsistencies in this list, so this additional
+				 * check is needed */
+				// TODO: See if inconsistencies might be thread-safeness related.
+				if (lastActivities.containsKey(playerUUID) && p != null) {
+					LastActivity la = entry.getValue();
+					la.loggedLocations.add(p.getLocation());
+					if (la.loggedLocations.size() >= maxLocations) {
+						if (la.loggedLocations.size() > maxLocations) {
+							la.loggedLocations.removeFirst();
+						}
+						// we tracking location even if on reprieve or immune, but that's it
+						if (!reprieve.containsKey(playerUUID) && !AFKPGC.immuneAccounts.contains(playerUUID)) {
+							int itWasntMeISwear = la.calculateMovementRadius();
+							if (itWasntMeISwear < smallestMovedDistance) {
+								smallestMovedDistance = itWasntMeISwear;
+								Player dirtyLiar = Bukkit.getPlayer(playerUUID);
+
+								topSuspects.put(itWasntMeISwear, new Suspect(
+										playerUUID, dirtyLiar.getName(), dirtyLiar.getLocation(),
+										la.evaluateBounds(relaxationFactor) ) );
+								AFKPGC.debug("Player ", playerUUID, " added as suspect (movement was ",
+										itWasntMeISwear, ")");
+
+								if (topSuspects.size() > maxSuspects) {
+									Suspect cleared = topSuspects.pollLastEntry().getValue(); // gets rid of largest distance
+									AFKPGC.debug("Player ", cleared.getUUID(), " released as suspect, better suspects found");
+								}
+							}
+						} else {
+							AFKPGC.debug("Skipping ", playerUUID, " due to reprieve or immunity");
+						}
+					} else {
+						AFKPGC.debug("Skipping ", playerUUID, " due to insufficient location data");
+					}
+				} else {
+					AFKPGC.debug("Player ", playerUUID, " likely offline.");
+				}
+			}
+			Suspect thisRoundSuspect = null;
+			// Now find first top suspect to pass the truebot tests.
+			if (topSuspects.size() > 0) {
+				for (Map.Entry<Integer, Suspect> entry : topSuspects.entrySet()) {
+					Suspect curSuspect = entry.getValue();
+					// Test Bounds for truebot(tm) detection.
+					BoundResults bounds = curSuspect.getResults();
+					if (bounds != null) {
+						double truebot = 0.0;
+						truebot += (bounds.getContained() ? boundsConfig.getContained() : 0.0);
+						truebot += (bounds.getContainedExcludeY() ? boundsConfig.getContainedExcludeY() : 0.0);
+						truebot += (bounds.getVolumeSimilar() ? boundsConfig.getVolumeSimilar() : 0.0);
+						truebot += (bounds.getSurfaceSimilar() ? boundsConfig.getSurfaceSimilar() : 0.0);
+						truebot += (bounds.getNearlyContained() ? boundsConfig.getNearlyContained() : 0.0);
+						truebot += (bounds.getNearlyContainedExcludeY() ? 
+								boundsConfig.getNearlyContainedExcludeY() : 0.0);
+
+						if (truebot >= boundsConfig.getThreshold()) {

```

Comment:
Shouldn't the total size of the box also be taken into account here? It's possible to slip through those checks with different boxes, which are each smaller than 10 blocks. Also how about using online time ?